Mathlib/Data/Finset/Image.lean

```diff
@@ -767,4 +767,25 @@ theorem finsetCongr_toEmbedding (e : α ≃ β) :
     e.finsetCongr.toEmbedding = (Finset.mapEmbedding e.toEmbedding).toEmbedding :=
   rfl
 
+/-- Given a predicate `p : α → Prop`, produces an equivalence between
+  `Finset {a : α // p a}` and `{s : Finset α // ∀ a ∈ s, p a}`. -/
+@[simps]
```

Comment:
Could oyu just copy and paste into a comment here what `simps` produces? I'm a little skeptical, without seeing code that relies on it, that it actually produces the most useful thigns here.

Reply:
Sure! @kim-em 

`Finset`s:
```lean
  @[simp]
  protected lemma finsetSubtypeCongr_apply_coe (p : α → Prop) (s : Finset {a // p a}) :
    (Equiv.finsetSubtypeCongr p) s = s.map ⟨Subtype.val, Subtype.val_injective⟩

  @[simp]
  protected lemma finsetSubtypeCongr_symm_apply (p : α → Prop) (s : { s // ∀ a ∈ s, p a }) :
    (Equiv.finsetSubtypeCongr p).symm s =
      s.val.attach.map (Subtype.impEmbedding (Membership.mem s.val) p fun _ ↦ s.property _)
```
`Set`s:
```lean
  @[simp]
  protected lemma setSubtypeCongr_apply_coe (p : α → Prop) (s : Set {a // p a}) (a : α) :
    (Equiv.setSubtypeCongr p) s a = ∃ (h : p a), s ⟨a, h⟩
  @[simp]
  protected lemma setSubtypeCongr_symm_apply
      (p : α → Prop) (s : {s // ∀ a ∈ s, p a}) (a : {a // p a}) :
    (Equiv.setSubtypeCongr p).symm s a = s.val a.val
```

Reply:
I don't think that Kim intended for these to be added in the file itself, just written out here for convenience (but of course, correct me if I'm wrong). It looks to me like the finset simp lemmas are exactly what I'd find useful, but the set ones aren't (in fact they're pretty bad!)

Reply:
Then I will write both simps lemma of Set separately.